serverlessworkflow/modules/ROOT/pages/service-orchestration/orchestration-of-openapi-based-services.adoc

```diff
@@ -0,0 +1,399 @@
+:understanding-jq-expressions: xref:core/understanding-jq-expressions.adoc
+:configuring-openapi-services-endpoints: xref:service-orchestration/configuring-openapi-services-endpoints.adoc
+:camel-k-integration: xref:integrations/camel-k-integration.adoc
+:authention-support-for-openapi-services: xref:security/authention-support-for-openapi-services.adoc
+
+This document describes how to call REST services via an link:{open_api_spec_url}[OpenAPI] specification file.
+
+The examples in this document assume that you have the Quarkus tooling installed and have created a Serverless Workflow project. 
+
+For more information about the tooling, see {getting-familiar-with-our-tooling}[Using Serverless Workflow tooling]. To create a new project, see {create-your-first-workflow-service}[Creating your first Serverless Workflow service].
+
+== The OpenAPI Function Definition
+
+Kogito follows the {spec_doc_url}#using-functions-for-restful-service-invocations[CNCF Serverless Workflow specification function definition] to make calls to remote REST services using an OpenAPI specification reference, as shown in the following example:
+
+.An example of an OpenAPI function definition
+[source,json]
+----
+{
+   "functions":[
+      {
+         "name":"myFunction1",
+         "operation":"file:///home/users/kogito/myopenapi-file.yaml#myFunction1"
+      }
+   ]
+}
+----
+
+The `operation` attribute is a URI composed by:
+
+1. The URI scheme that the engine uses to locate the specification file
+2. The location path of the resource
+3. The operation identifier (`operationId`), which you find in the link:{open_api_spec_url}#fixed-fields-7[OpenAPI specification file].
+
+Kogito supports the following URI schemes:
+
+* `classpath` for files located in the application project under the `src/main/resources` folder. The `classpath` is the default scheme when the you don't define a scheme in the URI. For example `classpath:/myopenapifile.yaml` or `/myopenapifile.yaml`. In this example, the file location is in `src/main/resources/myopenapifile.yaml` path
+* `file` for files located in the filesystem
+* `http` or `https` for remotely located files
+
+[IMPORTANT]
+====
+The OpenAPI specification files must be available only on build time. This is because, during runtime, Kogito leverages its internal code generation feature to make the REST calls. So {build-workflow-image-with-quarkus-cli}[once you build your application image], the Kogito engine won't access the files anymore.
+====
+
+== Getting Access to the OpenAPI Specification Files
+
+To get started, you must locate the REST services OpenAPI specification files.
+
+Usually, REST services expose their interface via the OpenAPI specification. For example, see the link:https://petstore3.swagger.io/api/v3/openapi.json[Petstore API].
+
+If the service you want to add to the workflow doesn't have the specification file, you can either create one or update the service to generate and expose it.
```

Comment:
I know you want people to generate the openAPI file, but I think is fair to tell them they have that possibility, since we are going to support it through custom operaiton type.

Reply:
I won't add in this guide, we can write a new one just tackling this and link here once it's done. JIRA: https://issues.redhat.com/browse/KOGITO-7413